addons/stock/models/stock_rule.py

```diff
@@ -35,36 +35,143 @@ class ProcurementRule(models.Model):
         ('fixed', 'Fixed')], string="Propagation of Procurement Group", default='propagate')
     group_id = fields.Many2one('procurement.group', 'Fixed Procurement Group')
     action = fields.Selection(
-        selection=[('move', 'Move From Another Location')], string='Action',
+        selection=[('pull', 'Pull From'), ('push', 'Push To'), ('pull_push', 'Pull & Push')], string='Action',
         required=True)
     sequence = fields.Integer('Sequence', default=20)
-    company_id = fields.Many2one('res.company', 'Company')
-    location_id = fields.Many2one('stock.location', 'Procurement Location')
-    location_src_id = fields.Many2one('stock.location', 'Source Location', help="Source location is action=move")
+    company_id = fields.Many2one('res.company', 'Company',
+        default=lambda self: self.env.user.company_id)
+    location_id = fields.Many2one('stock.location', 'Destination Location', required=True)
+    location_src_id = fields.Many2one('stock.location', 'Source Location')
     route_id = fields.Many2one('stock.location.route', 'Route', required=True, ondelete='cascade')
     procure_method = fields.Selection([
         ('make_to_stock', 'Take From Stock'),
-        ('make_to_order', 'Create Procurement')], string='Move Supply Method',
+        ('make_to_order', 'Trigger Another Rule')], string='Move Supply Method',
```

Comment:
is it possible to set a push rule mto? shouldn't we block this?

Reply:
Hide in view